src/main/java/com/theironyard/services/GuildWarsApiService.java

```diff
@@ -0,0 +1,97 @@
+package com.theironyard.services;
+
+import com.theironyard.entities.Item;
+import com.theironyard.entities.Price;
+import com.theironyard.entities.Recipe;
+import com.theironyard.repositories.ItemRepository;
+import com.theironyard.repositories.RecipeRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.*;
+
+@Service
+public class GuildWarsApiService {
+    public static final String RECIPES_URL = "https://api.guildwars2.com/v2/recipes/";
+    public static final String ITEMS_URL = "https://api.guildwars2.com/v2/items/";
+    public static final String PRICES_URL = "https://api.guildwars2.com/v2/commerce/prices/";
+    public static final String INGREDIENT_ITEM_ID = "item_id";
+    public static final String INGREDIENT_COUNT = "count";
+    public static final String ITEM_UNIT_PRICE = "unit_price";
+
+    @Autowired
+    RestTemplate restTemplate;
+
+    @Autowired
+    RecipeRepository recipeRepository;
+
+    @Autowired
+    ItemRepository itemRepository;
+
+    public GuildWarsApiService() {
+    }
+
+    public Recipe getRecipeById(String id){
+        Recipe recipe;
+
+        if (recipeRepository.exists(Integer.valueOf(id))){
```

Comment:
This does work.  However, you will be doing 2 database calls.  I would recommend trying to just pull the recipe and checking for null.  This way you will do 1 query since you will need the record anyway

Another way you might want to do caching is to store it in memory.  The database is obviously much faster than an API call but given the transient nature of the data you might as well just store it in a static hashmap.  it wouldn't be as persistent but it would be dramatically faster than even a database call.